packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,182 @@
+import React from "react";
+import {
+  Root as RadixRoot,
+  Indicator as RadixIndicator,
+  CheckboxProps as RadixCheckboxProps,
+} from "@radix-ui/react-checkbox";
+import { styled, theme } from "@localyze-pluto/theme";
+import { SystemProp, Theme } from "@xstyled/styled-components";
+import { Box } from "../../primitives/Box";
+import { Text } from "../../primitives/Text";
+import { Icon } from "../Icon";
+
+const StyledCheckboxIndicator = styled.span`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 14px;
+  height: 14px;
+  padding-bottom: ${theme.space.space10};
+
+  svg {
+    stroke-width: 2.5;
+  }
+`;
+
+export const CheckIndicator = (): React.ReactElement => {
+  return (
+    <StyledCheckboxIndicator as={RadixIndicator} data-testid="checkbox-check">
+      <Icon
+        color="colorBackground"
+        decorative={true}
+        icon="CheckIcon"
+        size="sizeIcon30"
+      />
+    </StyledCheckboxIndicator>
+  );
+};
+
+export const IndeterminateIndicator = (): React.ReactElement => {
+  return (
+    <StyledCheckboxIndicator
+      as={RadixIndicator}
+      data-testid="checkbox-indeterminate"
+    >
+      <Icon
+        color="colorBackground"
+        decorative={true}
+        icon="MinusSmallIcon"
+        size="sizeIcon30"
+      />
+    </StyledCheckboxIndicator>
+  );
+};
+
+export interface CheckboxProps extends RadixCheckboxProps {
+  /** Wrap or leave checkbox as is. */
+  wrapped?: boolean;
+  /** Provide error state of the checkbox. */
+  error?: boolean;
+  /** Checkbox id is required for proper functioning of htmlFor attribute on label. */
+  checkboxId: string;
```

Comment:
Why are we not using the native props from the Element?

Reply:
Because this props are also needed for the label and wrapping div which are outside of the checkbox

Reply:
If you check here https://github.com/radix-ui/primitives/blob/main/packages/react/checkbox/src/Checkbox.tsx#L35, you could get the id from the Button that Radix already uses to build the Checkbox.